controllers/storagecluster/reconcile.go

```diff
@@ -212,6 +212,9 @@ func (r *StorageClusterReconciler) validateStorageClusterSpec(instance *ocsv1.St
 		r.Log.Error(err, "Failed to validate StorageCluster version.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
 		r.recorder.ReportIfNotPresent(instance, corev1.EventTypeWarning, statusutil.EventReasonValidationFailed, err.Error())
 		instance.Status.Phase = statusutil.PhaseError
+		reason := statusutil.EventReasonValidationFailed
+		message := err.Error()
+		statusutil.SetVersionMismatchCondition(&instance.Status.Conditions, reason, message)
```

Comment:
Think about a case when there is no mismatch in the version you need to make it false in that case. This means you need to set it to false when the version verification is done.

Reply:
Oh, right. Thanks for the pointer.
Updated the code to set the status to False if the version check is successful.